ros/api/v1/hosts.py

```diff
@@ -25,30 +25,46 @@
 ]
 
 
+def default_queries(ident, additional_filter=None):
+    account_query = db.session.query(RhAccount.id).filter(RhAccount.account == ident['account_number']).subquery()
+    if additional_filter is None:
+        return db.session.query(System.id).filter(System.account_id.in_(account_query))
+    return db.session.query(System.id).filter(System.account_id.in_(account_query)).filter(additional_filter)
+
+
 class IsROSConfiguredApi(Resource):
     def get(self):
         ident = identity(request)['identity']
-        account_query = db.session.query(RhAccount.id).filter(
-            RhAccount.account == ident['account_number']).subquery()
-        system_count = db.session.query(System.id)\
-            .filter(System.account_id.in_(account_query)).count()
-        systems_with_suggestions = db.session.query(System.id)\
-            .filter(System.account_id.in_(account_query))\
-            .filter(System.number_of_recommendations > 0).count()
-        systems_waiting_for_data = db.session.query(System.id)\
-            .filter(System.account_id.in_(account_query))\
-            .filter(System.state == 'Waiting for data').count()
 
-        if system_count <= 0:
-            return {
-                'success': False,
-                'code': 'NOSYSTEMS',
-                'count': system_count
-            }
+        system_query = default_queries(ident)
 
+        last_reported = (
+            db.session.query(PerformanceProfile.system_id, func.max(PerformanceProfile.report_date).label('max_date')
+                             )
+            .filter(PerformanceProfile.system_id.in_(system_query.subquery()))
+            .group_by(PerformanceProfile.system_id)
+            .subquery()
+        )
+        print(last_reported)
```

Comment:
@saltgen need to remove the print statement. Forgot to do so. :)

Reply:
Np. I donno why I deleted the related comment :rofl: Oh well